target/linux/lantiq/dts/ALL0333CJ.dts

```diff
@@ -0,0 +1,131 @@
+/dts-v1/;
+
+#include "amazonse.dtsi"
+
+/ {
+	model = "ALL0333CJ - Allnet ALL0333CJ DSL Modem";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200";
+
+		aliases {
+			led-boot = &power;
+			led-failsafe = &power;
+			led-running = &power;
+
+			led-dsl = &dsl;
+			led-internet = &online_green;
+		};
+
+	};
+
+	memory@0 {
+		reg = <0x0 0x1000000>; // 16MB
+	};
+
+	// does not work at fpi@10000000 /TR
+	fpi@b0000000 {
+		etop@E180000 {
+			phy-mode = "mii";
+		};
+	};
+
+	fpi@10000000 {
+		gpio: pinmux@E100B10 {
+			pinctrl-names = "default";
+			pinctrl-0 = <&state_default>;
+
+			state_default: pinmux {
+				asc {
+					lantiq,groups = "asc";
+					lantiq,function = "asc";
+				};
+				keys_in {
+					lantiq,pins = "io0",/* "io25", */"io29";
+					lantiq,pull = <2>;
+					lantiq,open-drain = <1>;
+				};
+			};
+		};
+
+		// nor chips like: Spansion S29GL032
+		localbus@0 {
+			nor-boot@0 {
+				compatible = "lantiq,nor";
+				bank-width = <2>;
+				reg = <0 0x0 0x400000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					partition@0 {
+						label = "uboot";
+						reg = <0x00000 0x10000>;
+						read-only;
+					};
+
+					partition@10000 {
+						label = "firmware";
+						reg = <0x10000 0x3ef200>;
+					};
+
+					partition@3ff200 {
+						label = "uboot_env";
+						reg = <0x3ff200 0xc00>;
+						read-only;
+					};
+
+					partition@3ffe00 {
+						label = "dummy_rootfs";
+						reg = <0x3ffe00 0x200>;
+						read-only;
+					};
+				};
+			};
+		};
+	};
+
+	/**
+	 * gpios:
+	 *  1=dsl           (1=off, 0=on)
+	 *  3=ethernet      (1=off, 0=on)
+	 *  4=online red    (1=off, 0=on) - right most
+	 * 12=online green  (1=off, 0=on) - right most
+	 * 13=power         (1=red, 0=green) - left most
+	 */
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		/* POWER: two led's: 1x green, 1x red */
```

Comment:
ignore this one. failed to read your comment. The led is either green or red. But never off.